src/main/java/com/o19s/es/ltr/feature/store/StoredFeature.java

```diff
@@ -219,14 +218,16 @@ public Query doToQuery(LtrQueryContext context, FeatureSet set, Map<String, Obje
 
     private XContentParser createParser(Object source, NamedXContentRegistry registry) throws IOException {
         if (source instanceof String) {
-            return XContentFactory.xContent((String) source).createParser(registry, LoggingDeprecationHandler.INSTANCE, (String) source);
+            return MediaTypeRegistry.xContent((String) source).xContent()
```

Comment:
this xContent().xContent() call is really bugging me, but I definitely don't have another suggestion.

Reply:
The whole `MediaTypeRegistry.xContent` is deprecated (with the goal to not "guess" the content type)

Reply:
OK. so this is evolving. Are there other issues open in core or explanation of these deprecated methods that can be linked here so we can plan upgrades better?

Reply:
https://github.com/opensearch-project/OpenSearch/issues/8110